src/DIRAC/WorkloadManagementSystem/DB/PilotAgentsDB.py

```diff
@@ -629,6 +631,96 @@ def getPilotSummary(self, startdate='', enddate=''):
 #     return S_OK( result )
 
 ##########################################################################################
+  def getGroupedPilotSummary(self, selectDict, columnList):
```

Comment:
You should add a test for this in https://github.com/DIRACGrid/DIRAC/blob/integration/tests/Integration/WorkloadManagementSystem/Test_PilotAgentsDB.py